pkg/neutronapi/dbsync.go

```diff
@@ -73,15 +75,13 @@ func DbSyncJob(cr *neutronv1beta1.NeutronAPI, scheme *runtime.Scheme) *batchv1.J
 			},
 		},
 	}
-	initContainerDetails := common.InitContainer{
+	initContainerDetails := InitContainer{
 		ContainerImage: cr.Spec.ContainerImage,
-		DatabaseHost:   cr.Spec.DatabaseHostname,
+		DatabaseHost:   cr.Spec.DatabaseInstance,
 		Database:       Database,
-		NeutronSecret:  cr.Spec.NeutronSecret,
-		NovaSecret:     cr.Spec.NovaSecret,
+		NeutronSecret:  cr.Spec.Secret,
```

Comment:
Another approach we have been using for secrets is to use `XPasswordSelector` in the CR spec so that users can specify which keys in the secret contain the values needed for certain considerations: https://github.com/openstack-k8s-operators/keystone-operator/blob/038a3f1b65144cc53313cfb3a9952645a79f7a37/pkg/keystone/dbsync.go#L90-L92.  You then adjust the init container env var code to uses these selectors instead.

Reply:
Hmm i have used it in some places, will check and generalize to other places too.